components/org.wso2.carbon.identity.application.authenticator.adapter/src/main/java/org/wso2/carbon/identity/application/authenticator/adapter/internal/util/AuthenticatedUserBuilder.java

```diff
@@ -179,22 +181,54 @@ private User resolveLocalUserFromUserStore(String userId, AuthenticatedUserData.
         throw new ActionExecutionResponseProcessorException(errorMessage);
     }
 
-    private Map<ClaimMapping, String> resolveUserNameAndClaimsFromResponse() {
+    private Map<ClaimMapping, String> resolveUserNameAndClaimsFromResponse()
+            throws ActionExecutionResponseProcessorException {
 
         Map<ClaimMapping, String> userAttributes = new HashMap<>();
         for (AuthenticatedUserData.Claim claim : userFromResponse.getUser().getClaims()) {
+            validateClaimValue(claim.getUri(), claim.getValue());
             if (GROUP_CLAIM.equals(claim.getUri())) {
                 ignoreGroupsInClaimsInResponse();
                 continue;
+            } else if (ROLES_CLAIM.equals(claim.getUri())) {
+                continue;
             }
-            userAttributes.put(buildClaimMapping(claim.getUri()), claim.getValue());
+            userAttributes.put(buildClaimMapping(claim.getUri()), claim.getValueAsString());
             if (USERNAME_CLAIM.equals(claim.getUri())) {
-                usernameFromResponse = claim.getValue();
+                usernameFromResponse = claim.getValueAsString();
             }
         }
         return userAttributes;
     }
 
+    private void validateClaimValue(String claimUri, Object claimValue)
+            throws ActionExecutionResponseProcessorException {
+
+        if (ADDRESS_CLAIM.equals(claimUri)) {
+            return;
```

Comment:
Add a comment as ADDRESS claim is not validated for system defined characters like the multattribute separator as address is handled as a single valued claim internally

Reply:
This is not added right

Reply:
Already added https://github.com/wso2-extensions/identity-outbound-auth-adapter/pull/30/files#diff-7d0abb92a0e15d5cc6babc101fc178dd8c3cdb7d95af72f799eaf3ec41a8aab8R210-R212